### ISSUE TITLE
Export `typed-events` on `socket.io-client`

### ISSUE THREAD

**Issue by jzecca** [socketio/socket.io#3917]
**Is your feature request related to a problem? Please describe.**

v4 introduced support for typed events (and that's awesome 👍).
However, types can only be imported from `socket.io-client/build/typed-events`.

I'm actually not sure if this is intended or not.

**Describe the solution you'd like**
```ts
import { EventNames, EventParams, ReservedOrUserListener } from 'socket.io-client';
```

**Comment by MaximeKjaer** [comment]
Hi! I'm the person who contributed the typed events :)

I did not export the typed events, because I wanted to keep them as an internal implementation detail. I didn't want them to be added to the officially supported API, because that would mean that changes to these types could potentially be breaking changes for users depending on them. While typed events are a cool feature of socket.io 4.x, I think it's out of the scope of this project to provide a general-purpose typed `EventEmitter` implementation.

But at the same time, I wrote the typed event emitter code to be general enough to be applicable to other projects. I had a quick look at other typed `EventEmitter` libraries on npm, but did not find anything that met all the criteria below:

- Compatible with Node.js `EventEmitter`
- Different events for `on` and `emit`
- Reserved events, which:
  - can only be emitted by the library
  - can be listened to by users

At the time, I thought that since nothing of the sort existed, I could make it a library, and add it as a dependency of socket.io. However, I ended up deciding against that, because depending on an external library would make fixing things harder for the socket.io maintainers.

But if you have a use-case for a typed events library that fits the exact criteria above, and that nothing else like that exists, making a library out of the typed events code could be an option!
